packages/ui/src/Shared/DndFileChooser.tsx

```diff
@@ -0,0 +1,312 @@
+    return <Container />;
+  }
+
+  /**
+   * Callback for when one or more files were uploaded successfully.
+   * @param acceptedFiles the files that have been uploaded
+   */
+  public handleAcceptedFiles(acceptedFiles: File[]) {
+    this.setState({ ...this.state, files: acceptedFiles });
+    this.props.onUploadAccepted(acceptedFiles); // notify callback
+  }
+
+  /**
+   * Callback for when one or more files were rejected.
+   * @param rejectedFiles the files that were rejected
+   */
+  public handleRejectedFiles(rejectedFiles: File[]) {
+    const notifications = rejectedFiles.map(file => ({
+      key: file.name,
+      message: this.props.onUploadRejected(file.name),
+    }));
+
+    // If single file dropped then all files will be cleared. If multiple allowed, and multiple dropped,
+    // and some are accepted and some are rejected, the accepted files will still be uploaded because the
+    // handleAcceptedFiles is called after this method.
+    this.setState({
+      ...this.state,
+      files: [],
+      notifications: [...this.state.notifications, ...notifications],
+    });
+  }
+
+  /**
+   * Removes the specified notification from the list of notifications that have yet to be displayed.
+   * @param notificationToRemove the notificiation being removed
+   */
+  public removeNotificationAction(notificationToRemove: INotificationType) {
+    const notifications = this.state.notifications.filter(
+      notification => notificationToRemove.key !== notification.key
+    );
+    this.setState({
+      ...this.state,
+      notifications,
+    });
+  }
+
+  public render() {
+    return (
+      <Dropzone
+        accept={this.props.fileExtensions}
+        disabled={this.props.disableDropzone}
+        multiple={this.state.allowMultiple}
+        onDropAccepted={this.handleAcceptedFiles}
+        onDropRejected={this.handleRejectedFiles}
+      >
+        {({ getRootProps, getInputProps }) => (
+          <Container>
+            <ToastNotificationList className="dnd-file-chooser__notificationList">
+              {this.state.notifications.map(notification => (
+                <TimedToastNotification
+                  key={notification.key}
+                  type={'error'}
+                  persistent={false}
+                  onDismiss={this.removeNotificationAction.bind(
+                    this,
+                    notification
+                  )}
+                  timerdelay={this.state.notificationTimerDelay}
+                >
+                  <Container
+                    dangerouslySetInnerHTML={{
+                      __html: notification.message,
+                    }}
+                  />
+                </TimedToastNotification>
+              ))}
+            </ToastNotificationList>
+            <Grid
+              disabled={this.props.disableDropzone}
+              fluid={true}
+              className="dnd-file-chooser"
+              {...getRootProps({ refKey: 'dnd-file-chooser' })}
```

Comment:
Is this custom `refKey` required?

Reply:
It is required by the react-dropzone component. I'm not sure what it is used for as you can set it to anything you want. Not setting it does not break the behavior but does cause errors to show in the console.

Reply:
Alright, thanks!